include/exadg/operators/operator_base.cpp

```diff
@@ -2153,6 +2157,175 @@ OperatorBase<dim, Number, n_components>::evaluate_face_integrals() const
+    dealii::DynamicSparsityPattern dsp;
+    internal_init_system_matrix(tmp_matrix, dsp, dof_handler.get_communicator());
+    internal_calculate_system_matrix(tmp_matrix);
+
+    // collect the DoF indices of all cells
+    std::vector<std::vector<dealii::types::global_dof_index>> dof_indices_all_cells(
+      n_cells, std::vector<dealii::types::global_dof_index>(dofs_per_cell));
+    // and compute weights by counting the contributions to a DoF
+    initialize_dof_vector(weights);
+    for(unsigned int cell_batch_index = 0; cell_batch_index < matrix_free->n_cell_batches();
+        ++cell_batch_index)
+    {
+      unsigned int const n_filled_lanes =
+        matrix_free->n_active_entries_per_cell_batch(cell_batch_index);
+
+      for(unsigned int v = 0; v < n_filled_lanes; ++v)
+      {
+        auto const & cell = matrix_free->get_cell_iterator(cell_batch_index, v);
+
+        auto & dof_indices = dof_indices_all_cells[cell_batch_index * vectorization_length + v];
+        if(is_mg)
+          cell->get_mg_dof_indices(dof_indices);
+        else
+          cell->get_dof_indices(dof_indices);
+
+        for(auto const & i : dof_indices)
+          weights[i] += 1.;
+      }
+    }
+    weights.compress(dealii::VectorOperation::add);
+
+    // prepare the weights vector for symmetric weighting
+    for(unsigned int i = 0; i < weights.size(); ++i)
+    {
+      if(weights.in_local_range(i))
+        weights[i] = 1. / std::sqrt(weights[i]);
+    }
+    weights.update_ghost_values();
+
+    // cut out overlapped block matrices
+    std::vector<dealii::FullMatrix<Number>> overlapped_cell_matrices(
+      n_cells, dealii::FullMatrix<Number>(dofs_per_cell));
+
+    dealii::SparseMatrixTools::restrict_to_full_matrices<SparseMatrix,
+                                                         dealii::DynamicSparsityPattern,
+                                                         Number>(tmp_matrix,
+                                                                 dsp,
+                                                                 dof_indices_all_cells,
+                                                                 overlapped_cell_matrices);
+
+    // factorize and store cell matrices
+    matrices.resize(n_cells, LAPACKMatrix(dofs_per_cell));
+    for(unsigned int cell_batch_index = 0; cell_batch_index < matrix_free->n_cell_batches();
+        ++cell_batch_index)
+    {
+      unsigned int const n_filled_lanes =
+        matrix_free->n_active_entries_per_cell_batch(cell_batch_index);
+
+      for(unsigned int v = 0; v < n_filled_lanes; ++v)
+      {
+        // get overlapped cell matrix
+        auto const & overlapped_cell_matrix =
+          overlapped_cell_matrices[cell_batch_index * vectorization_length + v];
+
+        // store the cell matrix and renumber lexicographic
+        auto & lapack_matrix = matrices[cell_batch_index * vectorization_length + v];
+
+        auto const & lex = matrix_free->get_shape_info().lexicographic_numbering;
```

Comment:
MatrixFree can have multiple DoFHandlers, so shouldn't there somewhere appear the dof-index?

Reply:
see https://dealii.org/current/doxygen/deal.II/classMatrixFree.html#a0f344f8afc12a87625bf247c75730db1

Reply:
Fixed this and in another place 709ced2d727724f9d3795cf8dbf1540bf92ba552